encfs/RawFileIO.cpp

```diff
@@ -220,14 +221,17 @@ bool RawFileIO::write(const IORequest &req) {
   ssize_t bytes = req.dataLen;
   off_t offset = req.offset;
 
+  int eno = 0;
   while ((bytes != 0) && retrys > 0) {
+    errno = 0;
```

Comment:
I don't follow.. pwrite will return -1 if it sets errno.
So the only time you need to worry about errno is in the path "if (writeSize < 0)" which returns unconditionally.  So how can eno ever be != at line 247?

I'd rather not mutate global variables if not necessary.
You could rewrite as:  
eno = 0;
... pwrite
if (writeSize < 0) {
  eno = errno;
...
}

But then it's pretty clear that eno is only needed for that path.